.github/workflows/coverage.yml

```diff
@@ -0,0 +1,49 @@
+name: Coverage
+
+on:
+  schedule:
+    - cron: '0 0 * * 1'
+  push:
+    branches:
+      - main
+      - release-*
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  code-coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Enable Rust Caching
+        uses: Swatinem/rust-cache@v2
```

Comment:
According to https://github.com/marketplace/actions/rust-cache (first comment in example) the cache should come after the installation of the rust toolchain.

Reply:
49cafc31c94ca6ff54f5da806c8b9df670bd75b2